portfolio/src/main/java/com/google/sps/servlets/CommentsDatastore.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.*;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+public class CommentsDatastore {
+
+  public class Comment {
+    private String comment;
+    private long timestamp;
+
+    public Comment(String comment, long timestamp) {
+      this.comment = comment;
+      this.timestamp = timestamp;
+    }
+  } 
+
+    public CommentsDatastore() {}
+    
+    public List<Comment> fetchComments(int maxNumOfComments) {
+      Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      List<Entity> results = datastore.prepare(query)
+        .asList(FetchOptions.Builder.withLimit(maxNumOfComments));
```

Comment:
it seems you can set the limit directly on the query:
https://cloud.google.com/datastore/docs/concepts/queries#cursors_limits_and_offsets

```new Query().setLimit(maxNumOfComments).addSort(...)```

Reply:
I believe I'm using the appengine api for datastore instead of the cloud datastore, so the documentation is slightly different. I tried doing what you suggested above but it doesn't work in this case.

Reply:
got it, don't worry if that's the way the api works then just use what you have today.